pkg/lib/database/filter.go

```diff
@@ -435,6 +444,37 @@ func (e FilterEq) filterObject(v *types.Value) bool {
 	return false
 }
 
+type FilterHasPrefix struct {
+	Key, Prefix string
+}
+
+func (p FilterHasPrefix) FilterObject(s *types.Struct) bool {
+	val := pbtypes.Get(s, p.Key)
+	if strings.HasPrefix(val.GetStringValue(), p.Prefix) {
+		return true
+	}
+
+	list := val.GetListValue()
+	if list == nil {
+		return false
+	}
+
+	for _, v := range list.Values {
+		if strings.HasPrefix(v.GetStringValue(), p.Prefix) {
+			return true
+		}
+	}
+	return false
+}
+
+func (p FilterHasPrefix) AnystoreFilter() query.Filter {
+	f, err := query.ParseCondition(`{"` + p.Key + `":{"$regex": "^` + p.Prefix + `"}}`)
```

Comment:
Is it possible to use constructors from anystore like it is done in other places?

Reply:
Thank you, fixed!